data/spells/scripts/attack/envenom.lua

```diff
@@ -6,8 +6,8 @@ combat:setParameter(COMBAT_PARAM_DISTANCEEFFECT, CONST_ANI_EARTH)
 function onTargetCreature(creature, target)
 	local min = (creature:getLevel() / 80) + (creature:getMagicLevel() * 0.55) + 6
 	local max = (creature:getLevel() / 80) + (creature:getMagicLevel() * 0.75) + 7
-	local damage = math.random(math.floor(min), math.floor(max))
-	creature:addDamageCondition(target, CONDITION_POISON, 1, target:isPlayer() and math.floor(damage / 2 + 0.5) or damage)
+	local damage = math.random(math.floor(min) * 1000, math.floor(max) * 1000) / 1000
```

Comment:
It does not make sense. `min` and `max` are floating point values, and thus have precision of more than 3 digits certainly.

Multiplying by 1000, then dividing by 1000, is the same as not and just adds rounding errors (which do not matter anyway). The damage must be an integer in the end.

Since both ends of the range are floats, the random result can be anything within the range, with the precision you claim for.